packages/devtools_app/lib/src/screens/memory/panes/diff/controller/diff_pane_controller.dart

```diff
@@ -0,0 +1,66 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:collection/collection.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+import '../../../../../primitives/utils.dart';
+import '../../../../../shared/common_widgets.dart';
+import '../../../primitives/memory_utils.dart';
+import 'model.dart';
+
+class DiffPaneController {
+  final scrollController = ScrollController();
```

Comment:
can we create this scroll controller in the widget lifecycle methods? the FlatTable should have autoscroll to bottom functionality you can leverage

Reply:
I use ListView, that does not have this feature

Reply:
either way, we should be creating ScrollControllers within the widget lifecycle, there is still a way to scroll to the bottom without putting this scroll controller in business logic. We could add a listener to the snapshot list, and scroll to the bottom whenever a new snapshot has been added.